internal/controller/management/projectconfigs/project_configs.go

```diff
@@ -0,0 +1,258 @@
+	}
+	// Otherwise, requeue after a delay.
+	// TODO: Make the requeue delay configurable.
+	return ctrl.Result{RequeueAfter: 5 * time.Minute}, nil
+}
+
+func (r *reconciler) syncProjectConfig(
+	ctx context.Context,
+	projectConfig *kargoapi.ProjectConfig,
+) (kargoapi.ProjectConfigStatus, bool, error) {
+	logger := logging.LoggerFromContext(ctx)
+	status := projectConfig.Status.DeepCopy()
+
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReconciling,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Syncing",
+		Message:            "Ensuring project config webhook receivers",
+		ObservedGeneration: projectConfig.GetGeneration(),
+	})
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionFalse,
+		Reason:             "Syncing",
+		Message:            "Ensuring project config webhook receivers",
+		ObservedGeneration: projectConfig.GetGeneration(),
+	})
+
+	whReceivers, err := r.syncWebhookReceiversFn(ctx, projectConfig)
+	if err != nil {
+		logger.Error(err, "error ensuring webhook receivers")
+		conditions.Set(status, &metav1.Condition{
+			Type:               kargoapi.ConditionTypeReady,
+			Status:             metav1.ConditionFalse,
+			Reason:             "Syncing",
+			Message:            "Failed to ensure project config webhook receivers: " + err.Error(),
+			ObservedGeneration: projectConfig.GetGeneration(),
+		})
+		return *status, true, fmt.Errorf("error ensuring webhook receivers: %w", err)
+	}
+	status.WebhookReceivers = whReceivers
+
+	conditions.Delete(status, kargoapi.ConditionTypeReconciling)
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Synced",
+		Message:            "ProjectConfig is synced and ready for use",
+		ObservedGeneration: projectConfig.GetGeneration(),
+	})
+	return *status, false, nil
+}
+
+func (r *reconciler) syncWebhookReceivers(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+) ([]kargoapi.WebhookReceiver, error) {
+	logger := logging.LoggerFromContext(ctx)
+	if pc.Spec.WebhookReceivers == nil {
+		logger.Debug("ProjectConfig does not have any receiver configurations")
+		return nil, nil
+	}
+
+	logger.Debug("ensuring receivers",
+		"receiver-configs", len(pc.Spec.WebhookReceivers),
+	)
+
+	var webhookReceivers []kargoapi.WebhookReceiver
+	for _, rc := range pc.Spec.WebhookReceivers {
+		if rc.GitHub != nil {
+			whr, err := r.ensureGitHubWebhookReceiver(ctx, pc, rc)
+			if err != nil {
+				logger.Error(err, "error ensuring GitHub webhook receiver",
+					"receiver-config", rc,
+				)
+				return nil, fmt.Errorf("error ensuring GitHub webhook receiver: %w", err)
+			}
+			webhookReceivers = append(webhookReceivers, *whr)
+		}
+	}
+	return webhookReceivers, nil
+}
+
+func (r *reconciler) ensureGitHubWebhookReceiver(
```

Comment:
I think this function could be generic and work for all platforms we add support for, or it could even possibly go away...

The parts of this function that may vary by platform are:

1. How do we learn the name of the Secret we need to retrieve?
2. Once the Secret is retrieved, what field contains the actual value we want to use in calculating the path?

You only need a simple switch on which field isn't nil in pc to get the name of the Secret and the name of the key within it. Retrieve that secret, get the value for the key, and now you have everything you need to pass to `external.GenerateWebhookPath()`... and you won't have to implement a separate ensureXWebhookReceiver() for every new platform we support.